presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java

```diff
@@ -201,6 +206,9 @@ public void close()
                 try (PrestoStatement statement = new PrestoStatement(this)) {
                     statement.internalExecute("ROLLBACK");
                 }
+                for (QueryInterceptor queryInterceptor : this.queryInterceptorInstances) {
```

Comment:
I think for autocommit queries, this won't have a transaction ID.  So it could be done outside of the if check above.  Also, we should take care to handle exceptions.  Something like this:

https://github.com/prestodb/presto/blob/04138bc26a4fc801630fe936463d89d071d00c0a/presto-common/src/main/java/com/facebook/presto/common/block/ClosingBlockLease.java#L66-L84